src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -386,7 +388,12 @@ public boolean authenticate(final SecurityRequestChannel request) {
                 log.debug("User still not authenticated after checking {} auth domains", restAuthDomains.size());
             }
 
-            if (authCredentials == null && anonymousAuthEnabled) {
+            log.info(request.uri());
+            log.info(request.getHeaders());
+            if (authCredentials == null
+                && anonymousAuthEnabled
+                && checkIfRequestIsForAnonymousLogin(request.header("_auth_request_type_"))) {
```

Comment:
this block is currently causing Anonymous auth flow to fail for the request `GET /.kibana/_doc/config:3.0.0` since the header `_auth_request_type_` is not found. Fix, when available, will be on the front-end PR

Reply:
resolved by using dummy creds for anonymous user